track_and_trace/docs/troubleshooting_guide.md

```diff
@@ -0,0 +1,128 @@
+
+# Grid Track and Trace Testing Procedure #
+
+## Building Code and Docker Images ##
+
+The steps in this section cover building and testing Grid Track and Trace code
+and Docker images.
+
+### Build Local Docker Images ###
+
+First lets build the local Grid Docker Images used to run the compiled code
+passed in as a volume.
+
+```
+
+$ cd ../grid
+$ docker-compose -f docker-compose.yaml -f ../grid-contrib/track_and_trace/docker-compose.yaml build --force-rm
+```
+
+#### Verification Step ####
+
+To verify that this step has been completed correctly, run the following
+command:
+
+```
+$ docker images | grep grid
+```
+
+Checklist: Have the following Docker Images been created?
+
+* tnt-asset-client
+* grid_gridd
+* tnt-contract-builder
+* pike-contract-builder
+* schema-contract-builder
+* tnt-fish-client
+* tnt-server
+* tnt-shell
+* hyperledger/sawtooth-shell
+
+## Starting the Services ##
+
+The steps in this section cover creating the Grid Track and Trace containers
+and making sure that all of the services are communicating.
+
+### Create and Start Containers ###
+
+Now we will start the Grid containers, as well as the Track and Trace
+containers.
+
+From the Grid directory, run the following command:
+
+```
+$ docker-compose -f docker-compose.yaml -f ../grid-contrib/track_and_trace/docker-compose.yaml up
+```
+
+#### Verification Step ####
+
+To verify that this step has been completed correctly, run the following
+command:
+
+```
+$ docker ps
+```
+
+Checklist: Are the following Docker Containers running?
+
+* hyperledger/sawtooth-devmode-engine-rust:1.1
+* hyperledger/sawtooth-rest-api:1.1
+* hyperledger/sawtooth-settings-tp:1.1
+* hyperledger/sawtooth-validator:1.1
+* grid-track-and-trace-tp
+* grid_gridd
+* postgres
+* hyperledger/sawtooth-shell:1.1
+* grid-pike-tp
+* grid-schema-tp
+
+Next, verify that the Pike, Grid Schema, and Grid Track and Trace smart
+contracts have been committed to the blockchain. In the docker-compose logs,
+you should find the following messages:
+
+```
+pike-contract-builder      | ---------========= pike contract is loaded =========---------
+schema-contract-builder    | ---------========= grid schema contract is loaded =========---------
+tnt-contract-builder       | ---------========= track and trace contract is loaded =========---------
+```
+
+## Verify Communication and Functionality ##
+
+The steps in this section cover testing Grid Track and Trace using the
+AssetTrack demo application.
+
+### AssetTrack Test ###
+
+Finally, we will go through the AssetTrack demo application to ensure that the
+app is functioning properly and communicating with the Grid services.
+
+Complete the following actions:
+
+* Navigate to `localhost:8021` in a web browser.
```

Comment:
I think @chenette would tell you to make this a ordered list, as they are steps to be performed in a certain order.